client-libraries/java/rest-client/src/com/google/android/gcm/server/FirebaseSender.java

```diff
@@ -0,0 +1,25 @@
+package com.google.android.gcm.server;
+
+import static com.google.android.gcm.server.Constants.FCM_SEND_ENDPOINT;
+
+/**
+ * Created 25.11.2016
+ * @author Žana Hrastovšek
+ */
+public class FirebaseSender extends Sender {
```

Comment:
Instead of a different class, can you add a second constructor to the Sender class?
this second constructor can take an additional "String endpoint" argument.
This would make it more flexible if the user wants to point to a different endpoint (staging / test / etc)